diff -Nupr src.p5/sbr_main.c src.p6/sbr_main.c
--- src.p5/sbr_main.c	2010-10-21 13:06:37.000000000 +0300
+++ src.p6/sbr_main.c	2010-10-21 13:06:37.000000000 +0300
@@ -57,11 +57,17 @@ static const struct
   { 28000, 36000,  22050, 2,  7,  8, 2, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 32 kbit/s */
   { 36000, 44000,  22050, 2, 10,  9, 2, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 40 kbit/s */
   { 44000, 52000,  22050, 2, 12,  9, 3, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 48 kbit/s */
+  { 52000, 60000,  22050, 2, 12,  9, 3, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 56 kbit/s */
+  { 60000, 68000,  22050, 2, 14, 10, 3, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 64 kbit/s */
+  { 68000, 72001,  22050, 2, 14, 10, 3, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 72 kbit/s */
 
   { 24000, 28000,  24000, 2,  5,  6, 1, 0, -3, SBR_SWITCH_LRC, 3 }, /* nominal: 24 kbit/s */
   { 28000, 36000,  24000, 2,  7,  8, 2, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 32 kbit/s */
   { 36000, 44000,  24000, 2, 10,  9, 2, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 40 kbit/s */
   { 44000, 52000,  24000, 2, 12,  9, 3, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 48 kbit/s */
+  { 52000, 60000,  24000, 2, 12,  9, 3, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 56 kbit/s */
+  { 60000, 68000,  24000, 2, 14, 10, 3, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 64 kbit/s */
+  { 68000, 72001,  24000, 2, 14, 10, 3, 0, -3, SBR_SWITCH_LRC, 2 }, /* nominal: 72 kbit/s */
 
 };
 
