diff -Nupr src.p6/env_bit.h src.p7/env_bit.h
--- src.p6/env_bit.h	2010-10-21 13:06:37.000000000 +0300
+++ src.p7/env_bit.h	2010-10-21 13:06:37.000000000 +0300
@@ -22,12 +22,24 @@
 
 
 
-#define  SI_ID_BITS_AAC             3
-#define  SI_FILL_COUNT_BITS         4
-#define  SI_FILL_ESC_COUNT_BITS     8
-#define  SI_FILL_EXTENTION_BITS     4
-#define  ID_FIL                     6
-
+//#define  SI_ID_BITS_AAC             3
+//#define  SI_FILL_COUNT_BITS         4
+//#define  SI_FILL_ESC_COUNT_BITS     8
+//#define  SI_FILL_EXTENTION_BITS     4
+//#define  ID_FIL                     6
+typedef enum{
+  SI_ID_BITS                =(3),
+  SI_FILL_COUNT_BITS        =(4),
+  SI_FILL_ESC_COUNT_BITS    =(8),
+  SI_FILL_EXTENTION_BITS    =(4),
+  SI_FILL_NIBBLE_BITS       =(4),
+  SI_SCE_BITS               =(4),
+  SI_CPE_BITS               =(5),
+  SI_CPE_MS_MASK_BITS       =(2) ,
+  SI_ICS_INFO_BITS_LONG     =(1+2+1+6+1),
+  SI_ICS_INFO_BITS_SHORT    =(1+2+1+4+7),
+  SI_ICS_BITS               =(8+1+1+1),
+}SI_BITS;
 
 
 
diff -Nupr src.p6/stat_bits.c src.p7/stat_bits.c
--- src.p6/stat_bits.c	2010-10-21 13:06:37.000000000 +0300
+++ src.p7/stat_bits.c	2010-10-21 13:06:37.000000000 +0300
@@ -5,22 +5,6 @@
 
  /* the 3GPP instrumenting tool */
 
-typedef enum{
-  SI_ID_BITS                =(3),
-  SI_FILL_COUNT_BITS        =(4),
-  SI_FILL_ESC_COUNT_BITS    =(8),
-  SI_FILL_EXTENTION_BITS    =(4),
-  SI_FILL_NIBBLE_BITS       =(4),
-  SI_SCE_BITS               =(4),
-  SI_CPE_BITS               =(5),
-  SI_CPE_MS_MASK_BITS       =(2) ,
-  SI_ICS_INFO_BITS_LONG     =(1+2+1+6+1),
-  SI_ICS_INFO_BITS_SHORT    =(1+2+1+4+7),
-  SI_ICS_BITS               =(8+1+1+1),
-}SI_BITS;
-
-
-
 static int countMsMaskBits(int   sfbCnt,
                            int   sfbPerGroup,
                            int   maxSfbPerGroup,
